independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java

```diff
@@ -62,7 +61,7 @@ private double calculateScore(OriginCombination s) {
         for (OriginWithPreference o : s.getCollectedOrigins()) {
             combinationScore += Math.pow(extOrigins.size(),
                     highestRegistryPreference + 1 - o.getPreference().registryPreference)
-                    * ((double) (Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
+                    * (((double) (Integer.MAX_VALUE) + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```

Comment:
@aloubyansky I find this pretty confusing, do the parenthesis `(Integer.MAX_VALUE)` have any effect? I am not sure where the cast will have effect, could you make it a bit clearer with more/less parenthesis?

Reply:
Yes, it's casting Integer.MAX_VALUE to a double before adding 1.